### PR TITLE
[WIP] : Integrating datasource extension

### PR DESCRIPTION
**Pull request by sudaraka94**
### What does this PR do?
It ports the earlier version of datasource plugin to Che

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -55,6 +55,12 @@
                 <artifactId>jsr305</artifactId>
                 <version>${version.jsr305.plugin}</version>
             </dependency>
+            <!-- https://mvnrepository.com/artifact/mysql/mysql-connector-java -->
+            <dependency>
+                <groupId>mysql</groupId>
+                <artifactId>mysql-connector-java</artifactId>
+                <version>6.0.6</version>
```
Comment (riuvshin) [inline pom.xml:62]: all deps versions must be defined in https://github.com/eclipse/che-dependencies/blob/master/pom.xml

File: wsagent/che-datasource-api/pom.xml [hunk 2]
```diff
@@ -0,0 +1,167 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>che-agent-parent</artifactId>
+        <groupId>org.eclipse.che.core</groupId>
+        <version>5.16.0-SNAPSHOT</version>
```
Comment (riuvshin) [inline wsagent/che-datasource-api/pom.xml:7]: version is outdated please set actual version from master branch.

File: wsagent/che-datasource-ide/pom.xml [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    Contributors:
+      Codenvy, S.A. - initial API and implementation
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>che-agent-parent</artifactId>
+        <groupId>org.eclipse.che.core</groupId>
+        <version>5.16.0-SNAPSHOT</version>
```
Comment (riuvshin) [inline wsagent/che-datasource-ide/pom.xml:19]: version is outdated please set actual version from master branch.

File: wsagent/che-datasource-shared/pom.xml [hunk 4]
```diff
@@ -0,0 +1,92 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>che-agent-parent</artifactId>
+        <groupId>org.eclipse.che.core</groupId>
+        <version>5.16.0-SNAPSHOT</version>
```
Comment (riuvshin) [inline wsagent/che-datasource-shared/pom.xml:7]: version is outdated please set actual version from master branch.

File: wsagent/che-datasource-ssl-socket-factory/pom.xml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>che-agent-parent</artifactId>
+        <groupId>org.eclipse.che.core</groupId>
+        <version>5.16.0-SNAPSHOT</version>
```
Comment (riuvshin) [inline wsagent/che-datasource-ssl-socket-factory/pom.xml:7]: version is outdated please set actual version from master branch.

**Review by riuvshin — CHANGES_REQUESTED** [review]

**Comment by riuvshin** [comment]
ci-build

**Comment by skabashnyuk** [comment]
@sudaraka94 is this PR ready for review? Shall we start commenting?

**Comment by skabashnyuk** [comment]
FYI. mysql driver has uncompatible license and can't be added to Eclipse Che.

**Comment by sudaraka94** [comment]
@skabashnyuk I think I will correct the issues mentioned by @riuvshin first. And then open the PR again for review

**Comment by sunix** [comment]
as discussed, this PR is not ready to be merged

**Pull request closed without merging**